stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AccessibilityNodeInfoWrapper.java

```diff
@@ -100,4 +108,165 @@ public static String getIgnoredReasons(AccessibilityNodeInfoCompat node, View vi
+          actionLabels.add("click");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_LONG_CLICK:
+          actionLabels.add("long-click");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_ACCESSIBILITY_FOCUS:
+          actionLabels.add("accessibility-focus");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_CLEAR_ACCESSIBILITY_FOCUS:
+          actionLabels.add("clear-accessibility-focus");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_NEXT_AT_MOVEMENT_GRANULARITY:
+          actionLabels.add("next-at-movement-granularity");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_PREVIOUS_AT_MOVEMENT_GRANULARITY:
+          actionLabels.add("previous-at-movement-granularity");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_NEXT_HTML_ELEMENT:
+          actionLabels.add("next-html-element");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_PREVIOUS_HTML_ELEMENT:
+          actionLabels.add("previous-html-element");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_SCROLL_FORWARD:
+          actionLabels.add("scroll-forward");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_SCROLL_BACKWARD:
+          actionLabels.add("scroll-backward");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_CUT:
+          actionLabels.add("cut");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_COPY:
+          actionLabels.add("copy");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_PASTE:
+          actionLabels.add("paste");
+          break;
+        case AccessibilityNodeInfoCompat.ACTION_SET_SELECTION:
+          actionLabels.add("set-selection");
+          break;
+        default:
+          if (action.getLabel() != null) {
+            actionLabels.add(action.getLabel());
+          } else {
+            actionLabels.add("unknown");
+          }
+          break;
+      }
+    }
+
+    String actions = String.valueOf(actionLabels);
+    return actions.substring(1, actions.length() -1);
+  }
+
+  @Nullable
+  public static CharSequence getDescription(AccessibilityNodeInfoCompat node, View view) {
+    CharSequence contentDescription = node.getContentDescription();
+    CharSequence nodeText = node.getText();
+
+    boolean hasNodeText = !TextUtils.isEmpty(nodeText);
+    boolean isEditText = view instanceof EditText;
+
+    // EditText's prioritize their own text content over a contentDescription
+    if (!TextUtils.isEmpty(contentDescription) && (!isEditText || !hasNodeText)) {
+      return contentDescription;
+    }
+
+    if (hasNodeText) {
+      return nodeText;
+    }
+
+    // If there are child views and no contentDescription the text of all non-focusable children,
+    // comma separated, becomes the description.
+    if (view instanceof ViewGroup) {
+      String concatChildDescription = "";
```

Comment:
definitely use a `StringBuilder` here. Doing `+=` with a `String` in a loop is a classic performance anti-pattern for a language/runtime with immutable strings (like Java, C#/VB, etc.)